server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -236,11 +238,24 @@ public static class FileSettingsHealthIndicatorService implements HealthIndicato
             )
         );
 
+        public static final String DESCRIPTION_LENGTH_LIMIT_KEY = "fileSettings.descriptionLengthLimit";
+        static final Setting<Integer> DESCRIPTION_LENGTH_LIMIT = Setting.intSetting(
```

Comment:
Nit: Javadoc here that explains the "why" would be helpful context for future maintainers. e.g.: Why this is configurable, and under what conditions we'd want to set it.